src/platform/macos/view.rs

```diff
@@ -416,8 +425,27 @@ extern fn key_up(this: &Object, _sel: Sel, event: id) {
 
         state.last_insert = None;
 
+        // We need characters here to check for additional keys such as
+        // F21-F24.
+        state.raw_characters = {
```

Comment:
I'd prefer you just bind the characters to a local variable here, since using the `raw_characters` field (which is specifically used for special handling in `keyDown`) could cause confusion.